src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -1214,9 +1221,11 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			} else {
 				executeTaskResult = await this._executeTask(task, resolver, runSource);
 			}
+			this._inProgressTasks.delete(task._label);
```

Comment:
good candidate to go in a `finally` block.

Reply:
ooh nifty 🥇